File: swift/problem203.swift
public class ListNode {
    public var val: Int
    public var next: ListNode?
    public init(_ val: Int) {
        self.val = val
        self.next = nil
    }
}

class Solution {
    func removeElements(_ head: ListNode?, _ val: Int) -> ListNode? {
        var prev = head
        
        while (prev?.val == val) {
            prev = prev!.next
        }
        
        let head = prev
        var cur = prev?.next
        
        while (cur != nil) {
            if (cur!.val == val) {
                prev!.next = cur!.next
                cur = cur!.next
            } else {
                prev = cur!
                cur = cur!.next
            }
        }
        
        return head
    }
}

File: python/problem203.py
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        cur,fast = None,head
        while fast:
            if fast.val == val:
                if not cur:
                    fast,head = fast.next,fast.next
                else:
                    cur.next = fast.next
                    fast = fast.next
            else:
                cur = fast
                fast = fast.next
        return head
            

            
# https://leetcode.com/problems/remove-linked-list-elements/discuss/239179/Simple-Python-Solution-(No-Dummy-Variables)
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        rem, rem_tail = None, None
        while head:
            if head.val != val:
                temp = ListNode(head.val)
                if rem:  # not empty
                    rem_tail.next, rem_tail = temp, temp 
                else:
                    rem, rem_tail = temp, temp
            head = head.next
        return rem